antlr/Excellent2.g4

```diff
@@ -64,6 +53,18 @@ expression:
 	| expression op = (LTE | LT | GTE | GT) expression	# comparison
 	| expression op = (EQ | NEQ) expression				# equality
 	| expression AMPERSAND expression					# concatenation
-	| LPAREN expression RPAREN							# parentheses;
+	| TEXT												# textLiteral
+	| NUMBER											# numberLiteral
+	| TRUE												# true
+	| FALSE												# false
+	| NULL												# null;
+
+// a subset of expressions which can be followed by (), [] or .
```

Comment:
so syntactically you can no longer do `NULL.x` or `TRUE.x` but you can do `(upper)("x")` or `(contact).name`